client/selfrepair/normandy_driver.js

```diff
@@ -32,23 +31,73 @@ class LocalStorage {
 }
 
 /**
+ * EventEmitter for Heartbeat events. Unlike normal event emitters,
+ * Heartbeat events only occur once, and we guarantee that if a handler
+ * is registered _after_ an event occurs, it will still be executed.
+ */
+export class HeartbeatEmitter {
+  static EVENTS = [
+    'NotificationOffered',
+    'NotificationClosed',
+    'LearnMore',
+    'Voted',
+    'TelemetrySent',
+    'Engaged',
+  ]
```

Comment:
Do we care about the `SurveyExpired` or `WindowClosed` events? https://dxr.mozilla.org/mozilla-central/source/browser/components/uitour/UITour.jsm#1050-1052